viewing_party/party.py

```diff
@@ -1,23 +1,109 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+    if title and genre and rating: 
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+        return movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title in movie.values():
+            user_data["watchlist"].remove(movie)
+            user_data = add_to_watched(user_data, movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    movie_count = len(user_data["watched"])

```

Comment:
Since `movie_count` is only used once, I would consider removing the variable and directly calling `len(user_data["watched"])` if it doesn't impact readability much.